proof/refine/ARM/Tcb_R.thy

```diff
@@ -2346,62 +2346,64 @@ lemma setSchedulerAction_invs'[wp]:
   apply (simp add: ct_idle_or_in_cur_domain'_def)
   done
 
+(* FIXME RT: move to...? *)
+lemma as_user_valid_tcbs[wp]:
+  "as_user ptr f \<lbrace>valid_tcbs\<rbrace>"
+  unfolding as_user_def
+  apply wpsimp
+  apply (clarsimp simp: valid_tcbs_def get_tcb_ko_at)
+  apply (rename_tac s tcb a b)
+  apply (prop_tac "valid_tcb ptr tcb s")
+   apply blast
+  apply (clarsimp simp: valid_tcb_def valid_tcb_state_def obj_at_def is_reply_def
+                        valid_arch_tcb_def tcb_cap_cases_def)
+  done
+
 lemma tcbinv_corres:
  "tcbinv_relation ti ti' \<Longrightarrow>
   corres (dc \<oplus> (=))
          (einvs and simple_sched_action and Tcb_AI.tcb_inv_wf ti and current_time_bounded 1)
          (invs' and sch_act_simple and tcb_inv_wf' ti')
          (invoke_tcb ti) (invokeTCB ti')"
   apply (case_tac ti, simp_all only: tcbinv_relation.simps valid_tcb_invocation_def)
-         apply (rule corres_guard_imp [OF writereg_corres], simp+)[1]
-        apply (rule corres_guard_imp [OF readreg_corres], simp+)[1]
-       apply (rule corres_guard_imp [OF copyreg_corres], simp+)[1]
+          apply (rule corres_guard_imp[OF writereg_corres], simp+)[1]
+         apply (rule corres_guard_imp[OF readreg_corres], simp+)[1]
+        apply (rule corres_guard_imp[OF copyreg_corres], simp+)[1]
+       apply (clarsimp simp del: invoke_tcb.simps)
+       apply (rename_tac word a b fault_h time_h croot vroot ipcb sl' fault_h' time_h' croot'
+                         vroot' ipcb')
+       apply (rule_tac F="is_aligned word 5" in corres_req)
```

Comment:
5. Is there some special significance to this number? If the proof broke later on could we figure out what it was?

Reply:
Ah, right. The unsatisfying answer is "that's what the number was previously", but I might dig into it and see where it comes from.

Reply:
We will never know the true significance of 5, for it turns out it was unnecessary for the proof.

Reply:
good that this was sorted in that way.